src/ILLink.RoslynAnalyzer/TrimAnalysis/TrimAnalysisVisitor.cs

```diff
@@ -88,6 +104,35 @@ public override MultiValue VisitLiteral (ILiteralOperation literalOperation, Sta
 			return literalOperation.ConstantValue.Value == null ? NullValue.Instance : TopValue;
```

Comment:
Why don't we want to use a ConstIntValue for cases where the literal is an integer?

Reply:
I guess I just didn't get that far. When I do the constant checks, this should sort of take care of itself, but I'll add it here as well.